### ISSUE TITLE
doctor: false-positive 'External changes pending import' after any export

### ISSUE THREAD

**Issue by joyshmitz** [Dicklesworthstone/beads_rust#173]
## Bug

`br doctor` reports **"External changes pending import"** on a tracker that is fully in-sync — `br sync --status --json` confirms `dirty_count: 0`, `jsonl_newer: false`, `db_newer: false`, and the actual SHA256 of `issues.jsonl` matches `metadata.jsonl_content_hash` in `beads.db`.

## Root cause

`check_sync_metadata()` in `doctor.rs` (lines 1118–1124) implements its own staleness logic that has drifted from the canonical `compute_staleness()` in `sync/mod.rs` (lines 1814–1884). Three specific gaps:

### 1. Ignores `last_export_time`

```rust
// doctor.rs:1118 — only compares against last_import_time
let j_newer = last_import.as_ref().is_none_or(|import_time| {
    chrono::DateTime::parse_from_rfc3339(import_time).is_ok_and(|import_ts| {
        let import_sys_time = std::time::SystemTime::from(import_ts);
        jsonl_mtime.is_some_and(|m| m > import_sys_time)
    })
});
```

`sync/mod.rs` correctly uses `max(last_import_time, last_export_time)` (lines 1838–1854). After an export, JSONL mtime updates but `last_import_time` stays old, so `mtime > last_import_time` is always true → false positive.

### 2. Fetches but never uses content hash for staleness

`jsonl_hash` is fetched at doctor.rs:1087–1090 but only used for display, never for the staleness decision. `sync/mod.rs` (lines 1863–1867) correctly checks: even if mtime is newer, only report `jsonl_newer = true` when the content hash actually differs.

### 3. Missing mtime witness fast-path

`sync/mod.rs:1822–1832` has a fast-path using `METADATA_JSONL_MTIME` witness; doctor.rs doesn't use it at all.

## When it triggers

After **any** export operation:
- `br sync --flush-only`
- `br sync --merge`
- Any mutation command with auto-flush (`create`, `update`, `close`, `reopen`, etc.)

Timeline:
```
Time | Event                    | last_import | last_export | JSONL mtime | doctor j_newer
-----|--------------------------|-------------|-------------|-------------|---------------
T0   | Initial import           | T0          | —           | T0          | false ✓
T1   | Export (sync --flush)    | T0          | T1          | T1          | TRUE  ✗
T2   | Doctor runs              | T0          | T1          | T1          | TRUE  ✗

doctor.rs:  mtime(T1) > last_import(T0)                          = TRUE  (wrong)
sync/mod:   mtime(T1) > max(last_import(T0), last_export(T1))    = FALSE (correct)
```

## How the drift happened

`compute_staleness()` was correctly updated in commit `eb3d0c0` (2026-01-24) to handle both import/export times + hash verification. `check_sync_metadata()` in doctor.rs was never updated to match.

## Suggested fix

Either call `compute_staleness()` from doctor.rs instead of reimplementing the logic, or at minimum port the three missing checks (export time, content hash, mtime witness) from `sync/mod.rs:1838–1870` into `doctor.rs:1118–1124`.

## Environment

- br v0.1.26 (commit 674b9bd)
- macOS (Darwin 25.3.0)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

**Comment by Dicklesworthstone** [comment]
Fixed in f2e20d4.

Root cause: `check_sync_metadata()` in doctor.rs reimplemented staleness logic using only `last_import_time`, while the canonical `compute_staleness()` in sync/mod.rs correctly uses `max(last_import_time, last_export_time)` plus content hash verification.

Fix: doctor.rs now delegates to `compute_staleness()` directly instead of reimplementing the logic. All three gaps are addressed:
1. Export time is now considered
2. Content hash is used to confirm staleness (not just mtime)
3. Mtime witness fast-path is included

Falls back gracefully if `compute_staleness()` fails.
